File: ant-sdk/antd-php/src/Models/FilePutPublicResult.php
<?php

declare(strict_types=1);

namespace Autonomi\Antd\Models;

/**
 * Result of a public file upload. The DataMap is stored on-network as an
 * extra chunk; `$address` is the shareable retrieval handle.
 */
readonly class FilePutPublicResult
{
    public function __construct(
        /** Hex-encoded on-network DataMap address. */
        public string $address,
        /** Total storage cost paid in token units (atto). "0" if all chunks already existed. */
        public string $storageCostAtto,
        /** Total gas cost paid in wei as a decimal string. */
        public string $gasCostWei,
        /** Number of chunks stored on the network (uint64). */
        public int $chunksStored,
        /** Which payment mode was actually used: "auto", "merkle", or "single". */
        public string $paymentModeUsed,
    ) {}
}

File: ant-sdk/antd-java/src/main/java/com/autonomi/antd/models/FilePutPublicResult.java
package com.autonomi.antd.models;

/**
 * Result of a public file upload. The DataMap is stored on-network as an
 * extra chunk; {@code address} is the shareable retrieval handle.
 *
 * @param address          hex-encoded on-network DataMap address
 * @param storageCostAtto  total storage cost paid in atto tokens ("0" if all chunks already existed)
 * @param gasCostWei       total gas cost paid in wei as a decimal string
 * @param chunksStored     number of chunks stored on the network
 * @param paymentModeUsed  which payment mode was actually used
 */
public record FilePutPublicResult(
        String address,
        String storageCostAtto,
        String gasCostWei,
        long chunksStored,
        String paymentModeUsed) {}
